ports/stm32/machine_i2s.c

```diff
@@ -125,6 +125,13 @@ typedef struct _non_blocking_descriptor_t {
     bool copy_in_progress;
 } non_blocking_descriptor_t;
 
+typedef struct _pll_config_t {
+    int8_t bits;
+    int32_t rate;
+    uint16_t plli2sn;
+    uint16_t plli2sr;
+} pll_config_t;
```

Comment:
The table can be made smaller if these are packed more tightly, eg:
```c
uint32_t rate;
uint8_t bits;
uint8_t plli2sr;
uint16_t plli2sn;
```
You could then define a helper macro to add entries to the table in the right way:
```c
#define I2S_PLL_ENTRY(bits, rate, n, r) { rate, bits, r, n }
...
I2S_PLL_ENTRY(16, 44100, 302, 2),
...
```

Reply:
I updated the structure as per your suggestion.

Reply:
Looks good.